_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -504,6 +508,40 @@ public static function get_plans( $request ) {
 		return $data;
 	}
 
+	/**
+	 * Gets the products that are in use on wpcom.
+	 *
+	 * @param WP_REST_Request $request The request.
+	 *
+	 * @return object A JSON object of wpcom products if the request was sucessful, or an error code otherwise.
+	 */
+	public static function get_products( $request ) {
+		$wpcom_request = Client::wpcom_json_api_request_as_user(
+			'/products?_locale=' . get_user_locale(),
+			'v1.1',
+			array(
+				'method'  => 'GET',
+				'headers' => array(
+					'X-Forwarded-For' => Jetpack::current_user_ip( true ),
+				),
+			),
+			null,
+			'rest'
+		);
+
+		$response_code = wp_remote_retrieve_response_code( $wpcom_request );
+		if ( 200 === $response_code ) {
+			return json_decode( wp_remote_retrieve_body( $wpcom_request ) );
```

Comment:
This will currently retrieve a bunch of excess products that we'll never use (over 350 products!). Let's make sure to pick and return the ones we need and are going to use only.

Ideally, we should be limiting those on the WP.com endpoint (with a field or something), so we don't have to transfer them over the wire.

Reply:
Let's do this in another PR so that testing can start ASAP.